tensorflow/core/kernels/data/take_dataset_op.cc

```diff
@@ -67,7 +67,10 @@ int64 TakeDataset::Cardinality() const {
   }
   if (n == kInfiniteCardinality) {
     return count_;
+  } else if (count_ == kInfiniteCardinality) {
```

Comment:
When the input dataset has finite outputs and count=-1, Cardinality()
should return the cardinality of the input dataset instead of -1.